### PR TITLE
added new action type to button to toggle visibility of a target

### PR DESCRIPTION
**Pull request by rohitgupta274509**
Initially the text element is hidden.
image

When toggle the button the text element get visible.
image

Please review and approve it.

### CODE REVIEW

File: application/backend/src/main/java/ai/typerefinery/websight/models/components/forms/Button.java [hunk 1]
```diff
@@ -77,6 +77,11 @@ public class Button extends BaseFormComponent {
     @Default(values = "")
     public String target;
 
+    @Inject
+    @Getter
+    @Default(values = "")
+    public String hideButtonElementTarget;
```
Comment (Harikarthyk) [inline application/backend/src/main/java/ai/typerefinery/websight/models/components/forms/Button.java:83]: change this to toggleTarget

File: application/backend/src/main/resources/apps/typerefinery/components/clientlibs/clientlibs-header/.content.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
   "js": [
     "components.js",
     "modal.js",
-    "dropdown.js"
+    "dropdown.js",
+    "hidebuttonelement.js"
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/clientlibs/clientlibs-header/.content.json:10]: change to toggle.js

File: application/backend/src/main/resources/apps/typerefinery/components/clientlibs/clientlibs-header/hidebuttonelement.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+window.Typerefinery = window.Typerefinery || {};
+window.Typerefinery.HideButtonElement = Typerefinery.HideButtonElement || {};
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/clientlibs/clientlibs-header/hidebuttonelement.js:2]: Same ToggleComponent

File: application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/dialog/.content.json [hunk 4]
```diff
@@ -151,6 +151,11 @@
           "sling:resourceType": "wcm/dialogs/components/selectitem",
           "label": "Initial Theme",
           "value": "initialTheme"
+        },
+        "hideButtonElements": {
+          "sling:resourceType": "wcm/dialogs/components/selectitem",
+          "label": "Toggle Elements",
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/dialog/.content.json:157]: Can you please change the label to Toggle Components

File: application/backend/src/main/resources/apps/typerefinery/components/clientlibs/clientlibs-header/.content.json (hunk 2 above)
Comment (rohitgupta274509) [reply application/backend/src/main/resources/apps/typerefinery/components/clientlibs/clientlibs-header/.content.json]: Changed.

File: application/backend/src/main/java/ai/typerefinery/websight/models/components/forms/Button.java (hunk 1 above)
Comment (rohitgupta274509) [reply application/backend/src/main/java/ai/typerefinery/websight/models/components/forms/Button.java]: Changed.

File: application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js [hunk 5]
```diff
@@ -4,9 +4,10 @@ window.Typerefinery.Components.Forms = Typerefinery.Components.Forms || {};
 window.Typerefinery.Components.Forms.Button = Typerefinery.Components.Forms.Button || {};
 window.Typerefinery.Modal = Typerefinery.Modal || {};
 window.Typerefinery.Dropdown = Typerefinery.Dropdown || {};
+window.Typerefinery.ToggleComponent = Typerefinery.ToggleComponent || {};
 window.Typerefinery.Page.Theme = Typerefinery.Page.Theme || {};
 
-(function (ns, componentNs, modalNs, dropdownNs, themeNs, document, window) {
+(function (ns, componentNs, modalNs, dropdownNs, togglecomponentNs, themeNs, document, window) {
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js:10]: Please follow Camel case for the naming conversions.

File: application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js [hunk 6]
```diff
@@ -61,10 +60,12 @@ window.Typerefinery.Page.Theme = Typerefinery.Page.Theme || {};
                 dropdownNs.init($component, componentConfig);
             }else if(actionType === "initialTheme") {                
                 themeNs.init($component, componentConfig);
+            }else if(actionType === "toggleComponent") {               
+                togglecomponentNs.init($component, componentConfig);
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js:64]: same here as well.

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js (hunk 6 above)
Comment (rohitgupta274509) [reply application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js]: Changed.

File: application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js (hunk 5 above)
Comment (rohitgupta274509) [reply application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js]: Changed.

**Review by wildone — APPROVED** [review]

**Pull request merged**